### PR TITLE
chore(deps): bump node, yarn, eslint, typescript, versions PE-1252

### PR DESCRIPTION
**Pull request by fedellen**

### CODE REVIEW

File: src/utils/gateway_api.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ export class GatewayAPI {
 
 			this.lastError = resp.statusText ?? resp;
 		} catch (err) {
-			this.lastError = err instanceof Error ? err.message : err;
+			this.lastError = err instanceof Error ? err.message : ((err as unknown) as string);
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/utils/gateway_api.ts]: 2d269e3

**Review by dtfiedler — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -26,23 +26,22 @@
 		"@istanbuljs/nyc-config-typescript": "^1.0.1",
 		"@types/bn.js": "^5",
 		"@types/chai": "^4.2.15",
-		"@types/fs-extra": "^9.0.8",
 		"@types/jwk-to-pem": "^2.0.0",
 		"@types/lodash": "^4",
 		"@types/mime-types": "^2.1.0",
 		"@types/mocha": "^8.2.1",
-		"@types/node": "^14.14.32",
+		"@types/node": "^20.4.5",
```
Comment (arielmelendez) [inline package.json:33]: Is this the right types version for node 18?

File: src/utils/gateway_api.ts [hunk 3]
```diff
@@ -187,7 +187,7 @@ export class GatewayAPI {
 
 			this.lastError = resp.statusText ?? resp;
 		} catch (err) {
-			this.lastError = err instanceof Error ? err.message : err;
+			this.lastError = err instanceof Error ? err.message : (err as string);
```
Comment (arielmelendez) [inline src/utils/gateway_api.ts:190]: Perhaps better to toString or interpolate this?

**Review by arielmelendez — CHANGES_REQUESTED** [review]
Left some minor suggestions. However, I believe we need some README updates as a result of these changes.

File: package.json (hunk 2 above)
Comment (fedellen) [reply package.json]: 4a32b4f

File: src/utils/gateway_api.ts (hunk 3 above)
Comment (fedellen) [reply src/utils/gateway_api.ts]: e8261ef

**Review by arielmelendez — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -1,9 +1,12 @@
 {
 	"name": "ardrive-core-js",
-	"version": "1.19.0",
+	"version": "2.0.0",
 	"description": "ArDrive Core contains the essential back end application features to support the ArDrive CLI and Desktop apps, such as file management, Permaweb upload/download, wallet management and other common functions.",
 	"main": "./lib/exports.js",
 	"types": "./lib/exports.d.ts",
+	"engines": {
+		"node": ">=18.17.0"
```
Comment (fedellen) [inline package.json:8]: Any thoughts on limiting our users of Core to node 18.17.0?

**Pull request merged**